torch/_dynamo/variables/torch.py

```diff
@@ -324,6 +324,13 @@ def call_function(
             )
         elif self.value is torch.amp.autocast_mode.autocast:
             return AutocastModeVariable.create(target_values=args, kwargs=kwargs)
+        elif self.value in [torch.cuda.amp.autocast, torch.cpu.amp.autocast]:
+            assert "device_type" not in kwargs
```

Comment:
Should this be `unimplemented()` to graph break rather than assert?

Reply:
eager mode throwing error for this case:
```
Traceback (most recent call last):
  File "/scratch/ybliang/work/repos/pytorch/debug/debug2.py", line 29, in <module>
    print(fn(x))
  File "/scratch/ybliang/work/repos/pytorch/debug/debug2.py", line 20, in f2
    with torch.cuda.amp.autocast(device_type="cuda", dtype=torch.bfloat16):
TypeError: __init__() got an unexpected keyword argument 'device_type'
```
So Dynamo should throw ```AssertionError```?